src/shared-components/optionButton/style.ts

```diff
@@ -2,28 +2,43 @@ import styled from '@emotion/styled';
 import { css } from '@emotion/core';
 
 import { style as TYPOGRAPHY_STYLE } from '../typography';
-import { COLORS, ANIMATION, SPACER, BOX_SHADOWS } from '../../constants';
-import Container from '../container';
+import {
+  COLORS, ANIMATION, SPACER, BOX_SHADOWS, 
+} from '../../constants';
+import { containerStyles, ContainerType } from '../container/style';
 
-const getOptionTypeStyles = optionType => {
+import { OptionButtonProps } from '.';
+
+type BaseIconWrapperStylesProps = Required<
```

Comment:
This is just saying that `buttonType`, `optionType`, and `selected` from the `OptionButtonProps` are all required for the `BaseIconWrapperStylesProps`, yes?

Reply:
Somewhat 😬 

`OptionButtonProps` has `buttonType` and `selected` as potentially optional, so `Required` (a TS built-in Utility: https://www.typescriptlang.org/docs/handbook/utility-types.html#requiredt) casts it as non-optional. 🙂

So the type in this diff is equivalent to:

```
type BaseIconWrapperStylesProps = {
  buttonType: 'primary' | 'secondary';
  optionType: 'radio' | 'checkbox';
  selected: boolean;
}
```

But by using the Utilities with the imported type we can couple the style TypeScript types with the component TypeScript types. It's not necessary--if it were just one prop I wouldn't have done it--but by sharing types this explicitly I think it's easier to visualize what is being shared and spread elsewhere.

Reply:
I ended up writing the simpler type because while playing with this in the main app (under `v9.3-beta.0`) I realized that some of these styled components are being used directly, outside of the `OptionButton`. (Thankful for TypeScript for catching this.)